core/src/ldml/ldml_processor.cpp

```diff
@@ -497,5 +508,26 @@ ldml_processor::context_to_string(km_core_state *state, std::u32string &str, boo
     return ctxlen; // consumed the entire context buffer.
 }
 
+ldml_event_state::ldml_event_state(
+    km_core_state *s,
+    km_core_virtual_key v,
+    uint16_t m,
+    uint8_t i,
+    uint16_t e) {
+  this->state          = s;
+  this->vk             = v;
+  this->modifier_state = m;
+  this->is_key_down    = i;
+  this->event_flags    = e;
```

Comment:
Can we use the same parameter names as in the header (and matching member props)?

```suggestion
    km_core_state *state,
    km_core_virtual_key vk,
    uint16_t modifier_state,
    uint8_t is_key_down,
    uint16_t event_flags) {
  this->state          = state;
  this->vk             = vk;
  this->modifier_state = modifier_state;
  this->is_key_down    = is_key_down;
  this->event_flags    = event_flags;
```

Reply:
one of the platforms complained about shadowed parameters when I did that.

Reply:
ugh ok